test/integration/targets/var_precedence/test_var_precedence_playbook.yml

```diff
@@ -38,7 +38,7 @@
 
 - hosts: inven_overridehosts
   vars_files:
-    - "test_var_precedence.yml"
+    - "vars/test_var_precedence.yml"
```

Comment:
both versions should work as loader/dwim should try 'vars/' first

Reply:
Yep, I'm aware of this.  I've added it to my checklist so I don't forget to revisit.

Reply:
So this never actually worked that way (auto searching inside of `vars/`).  If you notice the other change I made, I renamed the playbook, and without `vars/` on it, it was loading the playbook as a vars file (the playbook and vars file were named the same), and we accept vars files that result in lists, so it was creating vars for `hosts`, `vars_files`, `roles`, etc...

I checked by debugging the result of the following line in devel

```
data = preprocess_vars(self._loader.load_from_file(vars_file, unsafe=True))
```

I had a previous bug that caused it to fail in this PR, where devel worked, but in the end the test in devel was loading the wrong file.

Reply:
k, open bug and i'll look at fixing, this 'should work that way'TM and we do have the correct data (play file location aka loader basedir) to do it right.

Reply:
#75526